File: TheMissingBeat2D/Assets/Scripts/MovementSteps.js
#pragma strict

private var beatTime : float = 1.0f;

private var beat : float = 0;
private var lastBeatTime : float = 0;

private var initPos : float = 0.0f;

function Start () {
	initPos = transform.position.x;
}

function Update () {
	//	Debug.Log((Time.time - lastBeatTime)/beatTime);
	transform.position.x = initPos + Mathf.Lerp((beat-1.0f)*2.0f, beat*2.0f, Mathf.Min(1.0f, (Time.time - lastBeatTime)/beatTime));
	rigidbody2D.velocity.x = 0.0f;
}

function Beat (b : int) {
	
}

function ContraBeat (b : int) {
	lastBeatTime = Time.time;
	beat = b;
}

function SetTimeBetweenBeats(t : float) {
	beatTime = t;
}

File: TheMissingBeat2D/Assets/Scripts/MovementSteps.cs
using UnityEngine;
using System.Collections;

public class MovementSteps : MonoBehaviour {
	private float beatTime = 1.0f;
	
	private float beat = 0;
	private float lastBeatTime = 0;
	
	private float initPos = 0.0f;
	
	private GameObject rendererObject;
	
	private int step = 2;
	
	private float rot;
	private float stepTime = 1.0f;
	
	public EaseType easeType;
	
	void Start () {
		initPos = transform.position.x;
		rendererObject = transform.GetChild(0).gameObject;
		
		rot = (float) step / (transform.lossyScale.x*Mathf.PI);
	}
	
	void Beat (int b) {
		
	}
	
	void ContraBeat (int b) {
		lastBeatTime = Time.time;
		beat = b;
		
		iTween.MoveBy(gameObject, iTween.Hash("x", step, "time", stepTime, "easeType", iTweenX.Ease(easeType), "oncomplete", "FixTweenPos"));
		iTween.RotateBy(rendererObject, iTween.Hash("z", -rot, "time", stepTime, "easeType", iTweenX.Ease(easeType)));
	}
	
	void SetTimeBetweenBeats(float t) {
		beatTime = t;
	}
	
	void FixTweenPos() {
		Vector3 wantedPos = transform.position;
		wantedPos.x = Mathf.Round(wantedPos.x/2.0f)*2.0f;
		
		rigidbody.MovePosition(wantedPos);
	}
	
	void SetBeatLength(float y) {
		stepTime = y;
	}
}